exchanges/stream/websocket.go

```diff
@@ -1004,3 +1010,31 @@ func checkWebsocketURL(s string) error {
 	}
 	return nil
 }
+
+// checkSubscriptions checks subscriptions against the max subscription limit
+// and if the subscription already exists.
+func (w *Websocket) checkSubscriptions(subs []ChannelSubscription) error {
+	if len(subs) == 0 {
+		return errNoSubscriptionsSupplied
+	}
+
+	w.subscriptionMutex.Lock()
+	defer w.subscriptionMutex.Unlock()
+
+	if w.MaxSubscriptionsPerConnection > 0 && len(w.subscriptions)+len(subs) > w.MaxSubscriptionsPerConnection {
+		return fmt.Errorf("%w: current subscriptions: %v, incoming subscriptions: %v, max subscriptions per connection: %v - please reduce enabled pairs",
+			errSubscriptionsExceedsLimit,
+			len(w.subscriptions),
+			len(subs),
+			w.MaxSubscriptionsPerConnection)
+	}
```

Comment:
optional: this could probably be below the range, if only in the situation someone sends an existing sub inside `subs` and is actually still within the limit. They'd still get an error, so I'm not super fussed